airbyte-workers/src/main/java/io/airbyte/workers/config/WorkerConfigurationBeanFactory.java

```diff
@@ -228,6 +228,9 @@ public WorkerConfigs checkWorkerConfigs(
                                           @Value("${airbyte.worker.job.kube.images.busybox}") final String busyboxImage,
                                           @Value("${airbyte.worker.job.kube.images.curl}") final String curlImage,
                                           @Named("jobDefaultEnvMap") final Map<String, String> jobDefaultEnvMap) {
+    if (useIsolatedNodeSelector && isolatedNodeSelectors.isEmpty()) {
```

Comment:
since this gets repeated so many times, I'd probably move it into a private method that takes in `useIsolatedNodeSelector` and `isolatedNodeSelectors` as arguments